compile_fx: make a compile event that corresponds to the fx_compile waitcounter

c00w opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #152983

This is a pretty minor change, but by having exact correspondence, we can
easily confirm data differences between perfetto and wait counters

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

c00w committed:
compile_fx: make a compile event that corresponds to the fx_compile waitcounter

This is a pretty minor change, but by having exact correspondence, we can
easily confirm data differences between perfetto and wait counters

[ghstack-poisoned]

masnesral commented on torch/_inductor/compile_fx.py:
```diff
@@ -772,6 +772,7 @@ def _compile_fx_inner(
 
     with (
         _WaitCounter("pytorch.wait_counter.fx_codegen_and_compile").guard() as _,
+        dynamo_timed("_compile_fx_inner.ods_log", log_pt2_compile_event=True),
```
I'd vaguely assume that any portion where we'd want to have a waitcounter would also want a dynamo_compile column. Should we be adding a dynamo_compile column for this? log_pt2_compile_event means we'll get one in pt2_compile_events.

Couple nits:
* I don't love the name of the key. This will show up in pt2_compile_events as "_compile_fx_inner.ods_log". It's already causing confusion that the same event is typically name differently among dynamo_compile, pt2_compile_events, ODS. Can we try to keep them in sync going forward?
* If you agree to making above change, maybe just use dynamo_timed for all of it (you added the waitcounter params after all)

c00w replied:
Renamed to fx_codegen_and_compile, let me know if you have a better suggestion.

re: dynamo_compile counters, I never use them, and prefer having timing only in pt2_compile_events. No strong objection though, if you want it as a column in dynamo_compile let me know and I can do it.

re: do it all through dynamo_timed - Ah nice, yeah I thought the format didn't match, and didn't realize we had added the override option. Updated :).

c00w committed:
Update on "compile_fx: make a compile event that corresponds to the fx_compile waitcounter"


This is a pretty minor change, but by having exact correspondence, we can
easily confirm data differences between perfetto and wait counters

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

c00w commented:
@pytorchbot merge

c00w committed:
Update on "compile_fx: make a compile event that corresponds to the fx_compile waitcounter"


This is a pretty minor change, but by having exact correspondence, we can
easily confirm data differences between perfetto and wait counters

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

c00w commented:
@pytorchbot merge

c00w commented:
@pytorchbot merge

c00w committed:
Update on "compile_fx: make a compile event that corresponds to the fx_compile waitcounter"


This is a pretty minor change, but by having exact correspondence, we can
easily confirm data differences between perfetto and wait counters

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

c00w commented:
@pytorchbot merge

c00w committed:
Update on "compile_fx: make a compile event that corresponds to the fx_compile waitcounter"


This is a pretty minor change, but by having exact correspondence, we can
easily confirm data differences between perfetto and wait counters

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

c00w commented:
@pytorchbot merge
